### PR TITLE
add notes/comments to pools

### PR DESCRIPTION
**Pull request by przemekk13**

### CODE REVIEW

**Comment by apmasell** [comment]
:+1:

File: miso-web/src/main/webapp/scripts/pool_ajax.js [hunk 1]
```diff
@@ -493,6 +493,61 @@ Pool.ui = {
 
   prepareElements : function () {
     Pool.ui.createElementSelectDatatable(jQuery('#platformType').val());
+  },
+
+  showPoolNoteDialog: function (poolId) {
+    var self = this;
+    jQuery('#addPoolNoteDialog')
```
Comment (frankie13) [inline miso-web/src/main/webapp/scripts/pool_ajax.js]: :crying_cat_face:

File: miso-web/src/main/webapp/pages/editPool.jsp [hunk 2]
```diff
@@ -195,6 +195,48 @@
...
+          <c:forEach items="${pool.notes}" var="note" varStatus="n">
+            <div class="exppreview" id="pool-notes-${n.count}">
+              <b>${note.creationDate}</b>: ${note.text}
+              <span class="float-right" style="font-weight:bold; color:#C0C0C0;">${note.owner.loginName}
+                <c:if test="${(project.securityProfile.owner.loginName eq SPRING_SECURITY_CONTEXT.authentication.principal.username)
+                                or fn:contains(SPRING_SECURITY_CONTEXT.authentication.principal.authorities,'ROLE_ADMIN')}">
+                <span style="color:#000000"><a href='#' onclick="Pool.ui.deletePoolNote('${pool.id}', '${note.noteId}');">
+                  <span class="ui-icon ui-icon-trash" style="clear: both; position: relative; float: right; margin-top: -15px;"/></a></span>
+                </c:if>
+              </span>
```
Comment (callunity) [inline miso-web/src/main/webapp/pages/editPool.jsp]: This closing `</span>` should be within the `<c:if>` tags

File: miso-web/src/main/webapp/pages/editPool.jsp [hunk 3]
```diff
@@ -195,6 +195,48 @@
...
+        </li>
+      </ul>
+      <c:if test="${fn:length(pool.notes) > 0}">
+        <div class="note" style="clear:both">
+          <c:forEach items="${pool.notes}" var="note" varStatus="n">
+            <div class="exppreview" id="pool-notes-${n.count}">
+              <b>${note.creationDate}</b>: ${note.text}
+              <span class="float-right" style="font-weight:bold; color:#C0C0C0;">${note.owner.loginName}
+                <c:if test="${(project.securityProfile.owner.loginName eq SPRING_SECURITY_CONTEXT.authentication.principal.username)
+                                or fn:contains(SPRING_SECURITY_CONTEXT.authentication.principal.authorities,'ROLE_ADMIN')}">
```
Comment (callunity) [inline miso-web/src/main/webapp/pages/editPool.jsp]: I'd prefer it if non-admin note-owners could delete their own notes (if they make a mistake in a note and can't delete it, then they'll end up creating another note to say "disregard the note above").

File: miso-web/src/main/webapp/scripts/pool_ajax.js [hunk 4]
```diff
@@ -493,6 +493,61 @@ Pool.ui = {
...
+      .html("<form>" +
+        "<fieldset class='dialog'>" +
+        "<label for='internalOnly'>Internal Only?</label>" +
+        "<input type='checkbox' checked='checked' name='internalOnly' id='internalOnly' class='text ui-widget-content ui-corner-all' />" +
+        "<br/>" +
+        "<label for='notetext'>Text</label>" +
+        "<input type='text' name='notetext' id='notetext' class='text ui-widget-content ui-corner-all' />" +
+        "</fieldset></form>");
```
Comment (callunity) [inline miso-web/src/main/webapp/scripts/pool_ajax.js]: Request: add `autofocus` attribute to the note text input.

**Comment by tonydebat** [comment]
Add warning to prevent adding an empty note.

**Pull request merged**